lattice.cpp is over register bound

```
/Users/paul/dev/music/shortcircuit3/libs/vembertech/vt_dsp/lattice.cpp:168:15: warning: array index 4 is past the end of the array (which contains 4 elements) [-Warray-bounds]
        __m128d in = d.reg[4];
                     ^     ~
/Users/paul/dev/music/shortcircuit3/libs/vembertech/vt_dsp/lattice.h:14:2: note: array 'reg' declared here
        __m128d reg[4];
        ^
/Users/paul/dev/music/shortcircuit3/libs/vembertech/vt_dsp/lattice.cpp:212:2: warning: array index 4 is past the end of the array (which contains 4 elements) [-Warray-bounds]
        d.reg[4] = in;
        ^     ~
/Users/paul/dev/music/shortcircuit3/libs/vembertech/vt_dsp/lattice.h:14:2: note: array 'reg' declared here
        __m128d reg[4];
        ^
```

is it even used though?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -172,7 +172,6 @@ set(VEMBERTECH_SOURCE
         libs/vembertech/vt_dsp/lattice.cpp
         libs/vembertech/vt_dsp/macspecific.cpp
         libs/vembertech/vt_dsp/basic_dsp.cpp
-        libs/vembertech/thread/threadsafety.cpp
         )
 
 set(SHORTCIRCUIT_INCLUDE_DIRS

--- a/libs/vembertech/vt_dsp/lattice.cpp
+++ b/libs/vembertech/vt_dsp/lattice.cpp
@@ -149,7 +149,7 @@ void iir_lattice_sd_x87(lattice_sd &d, double * __restrict Data, int NSamples)
 	}
 }
 
-// TODO måste testas
+// TODO mï¿½ste testas
 void iir_lattice_pd_serial_SSE2(lattice_pd &d, double * __restrict Data, int NSamples)
 {	
 	const __m128d v0_1 = _mm_set_sd(0.1);
@@ -165,7 +165,7 @@ void iir_lattice_pd_serial_SSE2(lattice_pd &d, double * __restrict Data, int NSa
 	__m128d V2 = d.v.V2;
 	__m128d V3 = d.v.V3;	
 	__m128d CG = d.v.ClipGain;
-	__m128d in = d.reg[4];
+	__m128d in = d.reg[3];
 
 	for(int i=0; i<NSamples; i++)
 	{					
@@ -209,7 +209,7 @@ void iir_lattice_pd_serial_SSE2(lattice_pd &d, double * __restrict Data, int NSa
 	d.reg[0] = r0;
 
 	d.reg[2] = regmult;
-	d.reg[4] = in;
+	d.reg[3] = in;
 }
 
 double map_resonance_to_2Qinv(double reso, double freq) 

--- a/src/loaders/akai_s6k_import.cpp
+++ b/src/loaders/akai_s6k_import.cpp
@@ -229,8 +229,6 @@ bool sampler::load_akai_s6k_program(const char *filename, char channel, bool rep
 
     akai_s6k_akp_kloc s6k_kloc[256];
 
-    assert(s6k_prg.n_keygroups <= 256);
-
     akai_s6k_akp_env s6k_ampenv[256];
     akai_s6k_akp_zone s6k_zone[256][4];
     bool zone_present[256][4];

--- a/src/loaders/akai_s6k_import.cpp
+++ b/src/loaders/akai_s6k_import.cpp
@@ -6,6 +6,11 @@
 //
 //-------------------------------------------------------------------------------------------------------
 
+#if WINDOWS
+#include <windows.h>
+#include <mmiscapi.h>
+#endif
+
 #include "globals.h"
 #include "infrastructure/logfile.h"
 #include "sampler.h"
@@ -17,9 +22,7 @@
 #include <string.h>
 #include <vt_dsp/basic_dsp.h>
 #include <vt_util/vt_string.h>
-#if WINDOWS
-#include <mmiscapi.h>
-#endif
+
 
 #include <algorithm>
 using std::max;

--- a/src/loaders/dls_import.cpp
+++ b/src/loaders/dls_import.cpp
@@ -82,7 +82,7 @@ struct giga_3ewa
     // fors�tt senare
 };
 
-#pragma pack()
+#pragma pack(pop)
 
 bool sample::parse_dls_sample(void *data, size_t filesize, unsigned int sample_id)
 {
@@ -378,7 +378,7 @@ bool sampler::parse_dls_preset(void *data, size_t filesize, char channel, int pa
             // no sub-regions, read as normal DLS
             char fn[256];
             int newzone;
-            sprintf(fn, "%s|%i", filename, wlnk.ulTableIndex);
+            sprintf(fn, "%s|%llu", filename, wlnk.ulTableIndex);
             if (add_zone(fn, &newzone, channel))
             {
                 sample_zone *z = &zones[newzone];

--- a/src/loaders/load_aiff.cpp
+++ b/src/loaders/load_aiff.cpp
@@ -92,7 +92,7 @@ struct aiff_INST_chunk
     aiff_loop releaseLoop;
 };
 
-#pragma pack
+#pragma pack(pop)
 
 bool sample::parse_aiff(void *data, size_t filesize)
 {

--- a/src/loaders/riff_wave.h
+++ b/src/loaders/riff_wave.h
@@ -92,7 +92,7 @@ struct wave_inst_chunk
     char vel_low, vel_high;
 };
 
-#pragma pack()
+#pragma pack(pop)
 
 // from MMsystem.h
 

--- a/src/loaders/sf2_import.cpp
+++ b/src/loaders/sf2_import.cpp
@@ -5,6 +5,9 @@
 //	Copyright 2004 Claes Johanson
 //
 //-------------------------------------------------------------------------------------------------------
+#if WINDOWS
+#include <windows.h>
+#endif
 #include "sf2_import.h"
 #include "globals.h"
 #include "infrastructure/logfile.h"

--- a/src/loaders/sfz_import.cpp
+++ b/src/loaders/sfz_import.cpp
@@ -5,6 +5,10 @@
 //	Copyright 2004 Claes Johanson
 //
 //-------------------------------------------------------------------------------------------------------
+#if WINDOWS
+#include <windows.h>
+#endif
+
 #include "globals.h"
 #include "sampler.h"
 #include "stdio.h"

--- a/src/sampler.cpp
+++ b/src/sampler.cpp
@@ -401,7 +401,7 @@ void sampler::ChannelController(char channel, int cc, int value)
 
 void sampler::purge_holdbuffer()
 {
-    cs_engine.enter();
+    std::lock_guard g(cs_engine);
     int z;
     list<int>::iterator iter = holdbuffer.begin();
     while (1)
@@ -419,8 +419,6 @@ void sampler::purge_holdbuffer()
         else
             iter++;
     }
-    cs_engine.leave();
-
     // note: m�ste remova entries n�r noter d�dar sig sj�lv auch
 }
 
@@ -550,12 +548,11 @@ bool sampler::get_key_name(char *str, int channel, int key)
 
 bool sampler::get_sample_id(const char *filename, int *s_id)
 {
-    cs_patch.enter();
+    std::lock_guard g(cs_patch);
     if (!filename)
     {
         if (s_id)
             *s_id = -1;
-        cs_patch.leave();
         return true; // finns redan, skulle man kunna s�ga, eftersom den inte ska laddas
     }
 
@@ -563,7 +560,6 @@ bool sampler::get_sample_id(const char *filename, int *s_id)
     {
         if (s_id)
             *s_id = atoi(filename + 6);
-        cs_patch.leave();
         return (samples[*s_id & (max_samples - 1)] != NULL);
     }
 
@@ -576,12 +572,10 @@ bool sampler::get_sample_id(const char *filename, int *s_id)
             {
                 if (s_id)
                     *s_id = s;
-                cs_patch.leave();
                 return true;
             }
         }
     }
-    cs_patch.leave();
     return false;
 }
 
@@ -718,7 +712,9 @@ bool sampler::add_zone(const TCHAR *filename, int *new_z, char part, bool use_ro
     }
     else
         s = -1;
-    cs_patch.enter();
+
+
+    std::lock_guard lockUntilEnd( cs_patch );
     InitZone(i);
 
     zones[i].part = part;
@@ -795,7 +791,6 @@ bool sampler::add_zone(const TCHAR *filename, int *new_z, char part, bool use_ro
         *new_z = i;
     zone_exists[i] = true;
     update_zone_switches(i);
-    cs_patch.leave();
     return true;
 }
 
@@ -808,12 +803,15 @@ bool sampler::replace_zone(int z, const char *filename)
     kill_notes(z);
     int s_old = zones[z].sample_id;
 
-    cs_patch.enter();
-    int s = GetFreeSampleId();
-    if (s < 0)
-        return false;
-    samples[s] = new sample(conf);
-    cs_patch.leave();
+    int s = 0;
+    {
+        std::lock_guard g(cs_patch);
+
+        s = GetFreeSampleId();
+        if (s < 0)
+            return false;
+        samples[s] = new sample(conf);
+    }
 
     if (!(samples[s]->load(filename)))
     {
@@ -822,8 +820,8 @@ bool sampler::replace_zone(int z, const char *filename)
         return false;
     }
 
-    cs_patch.enter();
 
+    std::lock_guard lockUntilEnd( cs_patch );
     if ((s_old >= 0) && samples[s_old]->forget())
     {
         delete samples[s_old];
@@ -893,7 +891,6 @@ bool sampler::replace_zone(int z, const char *filename)
     temp[31] = 0;
     vtCopyString(zones[z].name, temp, 32);
     update_zone_switches(z);
-    cs_patch.leave();
     return true;
 }
 
@@ -901,7 +898,7 @@ bool sampler::free_zone(uint32_t zoneid)
 {
     if (!zone_exists[zoneid])
         return false;
-    cs_patch.enter();
+    std::lock_guard g(cs_patch);
     kill_notes(zoneid);
     zone_exists[zoneid] = false;
     if ((zones[zoneid].sample_id >= 0) && samples[zones[zoneid].sample_id]->forget())
@@ -909,7 +906,6 @@ bool sampler::free_zone(uint32_t zoneid)
         delete samples[zones[zoneid].sample_id];
         samples[zones[zoneid].sample_id] = 0;
     }
-    cs_patch.leave();
     return true;
 }
 
@@ -1017,7 +1013,7 @@ void sampler::idle()
 
 void sampler::part_clear_zones(int p)
 {
-    cs_patch.enter();
+    std::lock_guard g(cs_patch);
     int i;
     for (i = 0; i < max_zones; i++)
     {
@@ -1026,7 +1022,6 @@ void sampler::part_clear_zones(int p)
             free_zone(i);
         }
     }
-    cs_patch.leave();
 }
 
 //-------------------------------------------------------------------------------------------------

--- a/src/sampler.h
+++ b/src/sampler.h
@@ -14,7 +14,7 @@ class sampler;
 #include "sampler_state.h"
 #include <list>
 #include <string>
-#include <thread/threadsafety.h>
+#include <thread>
 #include <vector>
 #include <vt_dsp/lipol.h>
 #include <vt_gui/vt_gui_constants.h>
@@ -256,7 +256,7 @@ class sampler
     bool editor_open;
     // AudioEffectX	*effect;
     multiselect *selected;
-    c_sec cs_patch, cs_gui, cs_engine;
+    std::mutex cs_patch, cs_gui, cs_engine;
     configuration *conf;
     char sample_replace_filename[256];
     char keystate[16][128];

--- a/src/sampler.h
+++ b/src/sampler.h
@@ -15,6 +15,7 @@ class sampler;
 #include <list>
 #include <string>
 #include <thread>
+#include <mutex>
 #include <vector>
 #include <vt_dsp/lipol.h>
 #include <vt_gui/vt_gui_constants.h>

--- a/src/sampler_notelogic.cpp
+++ b/src/sampler_notelogic.cpp
@@ -1,6 +1,3 @@
-#if WINDOWS
-#include <windows.h>
-#endif
 
 #include "sampler.h"
 #if !TARGET_HEADLESS
@@ -146,7 +143,6 @@ int sampler::GetFreeVoiceId(int group_id)
             polyphony--;
             v_free = 0;
         }
-        ::OutputDebugString("voice killed\n");
     }
 
     return v_free;

--- a/src/sampler_notelogic.cpp
+++ b/src/sampler_notelogic.cpp
@@ -1,3 +1,7 @@
+#if WINDOWS
+#include <windows.h>
+#endif
+
 #include "sampler.h"
 #if !TARGET_HEADLESS
 #include "shortcircuit_editor2.h"

--- a/src/sampler_process.cpp
+++ b/src/sampler_process.cpp
@@ -222,63 +222,64 @@ void sampler::process_audio()
         clear_block(output_part[op], block_size_quad);
 
     // clear buffers & process controls
-    cs_patch.enter();
-
-    // render voices
-    for (unsigned int v = 0; v < highest_voice_id; v++)
     {
-        if (voice_state[v].active) // is bottleneck at idle (?)
+        std::lock_guard g(cs_patch);
+
+        // render voices
+        for (unsigned int v = 0; v < highest_voice_id; v++)
         {
-            float *outbuf[3][2];
-            outbuf[0][0] =
-                get_output_pointer(voices[v]->zone->aux[0].output, 0, voices[v]->zone->part);
-            outbuf[0][1] =
-                get_output_pointer(voices[v]->zone->aux[0].output, 1, voices[v]->zone->part);
-            outbuf[1][0] =
-                get_output_pointer(voices[v]->zone->aux[1].output, 0, voices[v]->zone->part);
-            outbuf[1][1] =
-                get_output_pointer(voices[v]->zone->aux[1].output, 1, voices[v]->zone->part);
-            outbuf[2][0] =
-                get_output_pointer(voices[v]->zone->aux[2].output, 0, voices[v]->zone->part);
-            outbuf[2][1] =
-                get_output_pointer(voices[v]->zone->aux[2].output, 1, voices[v]->zone->part);
-
-            voice_state[v].active = voices[v]->process_block(
-                outbuf[0][0], outbuf[0][1], outbuf[1][0], outbuf[1][1], outbuf[2][0], outbuf[2][1]);
-
-            if (!voice_state[v].active)
+            if (voice_state[v].active) // is bottleneck at idle (?)
             {
-                polyphony--;
-                holdbuffer.remove(v);
+                float *outbuf[3][2];
+                outbuf[0][0] =
+                    get_output_pointer(voices[v]->zone->aux[0].output, 0, voices[v]->zone->part);
+                outbuf[0][1] =
+                    get_output_pointer(voices[v]->zone->aux[0].output, 1, voices[v]->zone->part);
+                outbuf[1][0] =
+                    get_output_pointer(voices[v]->zone->aux[1].output, 0, voices[v]->zone->part);
+                outbuf[1][1] =
+                    get_output_pointer(voices[v]->zone->aux[1].output, 1, voices[v]->zone->part);
+                outbuf[2][0] =
+                    get_output_pointer(voices[v]->zone->aux[2].output, 0, voices[v]->zone->part);
+                outbuf[2][1] =
+                    get_output_pointer(voices[v]->zone->aux[2].output, 1, voices[v]->zone->part);
+
+                voice_state[v].active =
+                    voices[v]->process_block(outbuf[0][0], outbuf[0][1], outbuf[1][0], outbuf[1][1],
+                                             outbuf[2][0], outbuf[2][1]);
+
+                if (!voice_state[v].active)
+                {
+                    polyphony--;
+                    holdbuffer.remove(v);
 #if TARGET_VST2
-                if (editor && editor->isOpen())
-                    track_zone_triggered(voice_state[v].zone_id, false);
+                    if (editor && editor->isOpen())
+                        track_zone_triggered(voice_state[v].zone_id, false);
 #endif
+                }
+            }
+        }
+
+        // process parts
+        for (int p = 0; p < n_sampler_parts; p++)
+        {
+            process_part(p);
+        }
+
+        process_global_effects();
+
+        // Process preview player
+        if (mpPreview->mActive)
+        {
+            bool ContinuePreviewing =
+                mpPreview->mpVoice->process_block(output[0], output[1], NULL, NULL, NULL, NULL);
+
+            if (!ContinuePreviewing)
+            {
+                mpPreview->SetPlayingState(false);
             }
         }
     }
-
-    // process parts
-    for (int p = 0; p < n_sampler_parts; p++)
-    {
-        process_part(p);
-    }
-
-    process_global_effects();
-
-    // Process preview player
-    if (mpPreview->mActive)
-    {
-        bool ContinuePreviewing =
-            mpPreview->mpVoice->process_block(output[0], output[1], NULL, NULL, NULL, NULL);
-
-        if (!ContinuePreviewing)
-        {
-            mpPreview->SetPlayingState(false);
-        }
-    }
-
-    cs_patch.leave();
 
     processVUs();
     // post amplitude

--- a/src/sampler_voice.cpp
+++ b/src/sampler_voice.cpp
@@ -6,18 +6,19 @@
 //
 //-------------------------------------------------------------------------------------------------
 
+#include <cassert>
+
 #include "sampler_voice.h"
 #include "controllers.h"
 #include "sampler.h"
 #include "util/tools.h"
-#include <fstream>
 // #include <intrin.h>
 #include "generator.h"
 #include "synthesis/mathtables.h"
 #include "sample.h"
 #include "sampler_state.h"
 #include "synthesis/filter.h"
-#include <assert.h>
+
 #include <vt_dsp/basic_dsp.h>
 #include <vt_dsp/halfratefilter.h>
 
@@ -39,8 +40,8 @@ sampler_voice::sampler_voice(uint32 voice_id, timedata *td)
     halfrate = (halfrate_stereo *)_mm_malloc(sizeof(halfrate_stereo), 16);
     new (halfrate) halfrate_stereo(4, false);
 
-    voice_filter[0] = 0;
-    voice_filter[1] = 0;
+    voice_filter[0] = nullptr;
+    voice_filter[1] = nullptr;
 
     this->voice_id = voice_id;
     this->td = td;
@@ -111,9 +112,9 @@ sampler_voice::sampler_voice(uint32 voice_id, timedata *td)
 sampler_voice::~sampler_voice()
 {
     spawn_filter_release(voice_filter[0]);
-    voice_filter[0] = 0;
+    voice_filter[0] = nullptr;
     spawn_filter_release(voice_filter[1]);
-    voice_filter[1] = 0;
+    voice_filter[1] = nullptr;
 
     halfrate->~halfrate_stereo();
     _mm_free(halfrate);
@@ -143,7 +144,7 @@ void sampler_voice::play(sample *wave, sample_zone *zone, sample_part *part, uin
     part->last_note = key;
 
     this->velocity = velocity;
-    this->channel = channel;
+    // Channel is not an input - so comment this out this->channel = channel;
     this->ctrl = ctrl;
     this->automation = autom;
 
@@ -165,8 +166,8 @@ void sampler_voice::play(sample *wave, sample_zone *zone, sample_part *part, uin
     slice_env = 0;
     time60 = 0;
     loop_pos = 0;
-    random = (float)rand() / RAND_MAX;
-    randombp = ((float)rand() / RAND_MAX) * 2.f - 1.f;
+    random = (float)rand() / (float)RAND_MAX;
+    randombp = ((float)rand() / (float)RAND_MAX) * 2.f - 1.f;
     envelope_follower = 0;
     grain_id = 0;
 
@@ -184,7 +185,7 @@ void sampler_voice::play(sample *wave, sample_zone *zone, sample_part *part, uin
 
     halfrate->reset();
 
-    mm.assign(0, zone, part, this, ctrl, autom, td);
+    mm.assign(nullptr, zone, part, this, ctrl, autom, td);
     mm.process();
 
     first_run = true;
@@ -214,8 +215,8 @@ void sampler_voice::play(sample *wave, sample_zone *zone, sample_part *part, uin
 
     last_ft[0] = 0;
     last_ft[1] = 0;
-    voice_filter[0] = 0;
-    voice_filter[1] = 0;
+    voice_filter[0] = nullptr;
+    voice_filter[1] = nullptr;
 
     check_filtertypes();
 
@@ -300,7 +301,7 @@ void sampler_voice::play(sample *wave, sample_zone *zone, sample_part *part, uin
     }
 
     // choose generator function pointer
-    Generator = 0;
+    Generator = nullptr;
 
     int gmode = 0;
     switch (playmode)
@@ -372,7 +373,7 @@ void sampler_voice::uberrelease()
     is_uberrelease = true;
 }
 
-inline int sampler_voice::get_filter_type(int id) { return zone->Filter[id].type; }
+inline int sampler_voice::get_filter_type(int id) const { return zone->Filter[id].type; }
 
 void sampler_voice::check_filtertypes()
 {
@@ -381,7 +382,7 @@ void sampler_voice::check_filtertypes()
     {
         spawn_filter_release(voice_filter[0]);
         voice_filter[0] = spawn_filter(get_filter_type(0), mm.get_destination_ptr(md_filter1prm0),
-                                       zone->Filter[0].ip, 0, true);
+                                       zone->Filter[0].ip, nullptr, true);
         if (voice_filter[0])
             voice_filter[0]->init();
         last_ft[0] = get_filter_type(0);
@@ -390,7 +391,7 @@ void sampler_voice::check_filtertypes()
     {
         spawn_filter_release(voice_filter[1]);
         voice_filter[1] = spawn_filter(get_filter_type(1), mm.get_destination_ptr(md_filter2prm0),
-                                       zone->Filter[1].ip, 0, true);
+                                       zone->Filter[1].ip, nullptr, true);
         if (voice_filter[1])
             voice_filter[1]->init();
         last_ft[1] = get_filter_type(1);
@@ -434,9 +435,7 @@ void sampler_voice::update_portamento()
 #if perfprof
 #define perfslot(x) perf[x] = __rdtsc();
 #else
-#define perfslot(x)                                                                                \
-    {                                                                                              \
-    }
+#define perfslot(x)
 #endif
 
 void sampler_voice::CalcRatio()
@@ -469,27 +468,30 @@ bool sampler_voice::process_block(float *p_L, float *p_R, float *p_aux1L, float 
 #endif
     int VE = zone->element_active;
 
-    perfslot(0)
+    perfslot(0);
 
-        check_filtertypes();
+    check_filtertypes();
 
-    perfslot(1)
+    perfslot(1);
 
-        // process envelopes & stepLFO's
-        bool continue_playing = AEG.Process(block_size);
+    // process envelopes & stepLFO's
+    bool continue_playing = AEG.Process(block_size);
 
     if (VE & ve_EG2)
         EG2.Process(block_size);
-    perfslot(2) if (VE & ve_LFO1) stepLFO[0].process(block_size);
+    perfslot(2);
+    if (VE & ve_LFO1)
+        stepLFO[0].process(block_size);
     if (VE & ve_LFO2)
         stepLFO[1].process(block_size);
     if (VE & ve_LFO3)
         stepLFO[2].process(block_size);
 
-    perfslot(3) mm.process();
-    perfslot(4)
+    perfslot(3);
+    mm.process();
+    perfslot(4);
 
-        if (first_run)
+    if (first_run)
     {
         pfg.set_target_instantize(db_to_linear(mm.get_destination_value(md_prefilter_gain)));
         float amp = crossfade_amp * db_to_linear((1 - fvelocity) * zone->velsense) * AEG.output;
@@ -541,9 +543,10 @@ bool sampler_voice::process_block(float *p_L, float *p_R, float *p_aux1L, float 
         fmix2.set_target_smoothed(limit_range(mm.get_destination_value(md_filter2mix), 0.f, 1.f));
     }
 
-    perfslot(5)
+    perfslot(5);
 
-        if (portamento_active) update_portamento();
+    if (portamento_active)
+        update_portamento();
     CalcRatio();
     if (use_oversampling)
         GD.Ratio = GD.Ratio >> 1;
@@ -553,11 +556,11 @@ bool sampler_voice::process_block(float *p_L, float *p_R, float *p_aux1L, float 
 
     int bs = GD.BlockSize;
 
-    perfslot(6)
+    perfslot(6);
 
-        // GD.RatioMask = looping_active ? 0xffffffff : 0;
+    // GD.RatioMask = looping_active ? 0xffffffff : 0;
 
-        if (looping_active)
+    if (looping_active)
     {
         int end = (int)wave->sample_length;
         int ll = mm.get_destination_value_int(md_loop_length);
@@ -578,42 +581,43 @@ bool sampler_voice::process_block(float *p_L, float *p_R, float *p_aux1L, float 
         }
     }
 
-    perfslot(7)
+    perfslot(7);
 
-        if (use_stereo) pfg.multiply_2_blocks(
-            output[0], output[1], use_oversampling ? (block_size_quad << 1) : block_size_quad);
-    else pfg.multiply_block(output[0], use_oversampling ? (block_size_quad << 1) : block_size_quad);
+    if (use_stereo)
+        pfg.multiply_2_blocks(output[0], output[1],
+                              use_oversampling ? (block_size_quad << 1) : block_size_quad);
+    else
+        pfg.multiply_block(output[0], use_oversampling ? (block_size_quad << 1) : block_size_quad);
 
     if (use_oversampling)
     {
         halfrate->process_block_D2(output[0], output[1], block_size << 1);
     }
 
-    perfslot(8)
+    perfslot(8);
 
-        // envelope follower
-        /*{
-                const float integratorconst = samplerate_inv * block_size;
-                float bs_inv = (float)1/bs;
-                float ef_newvalue = sqrt(block_rms*bs_inv);
-                if (stereo) ef_newvalue *= 0.5;
-                float ef_rate;
+    // envelope follower
+    /*{
+            const float integratorconst = samplerate_inv * block_size;
+            float bs_inv = (float)1/bs;
+            float ef_newvalue = sqrt(block_rms*bs_inv);
+            if (stereo) ef_newvalue *= 0.5;
+            float ef_rate;
 
-                if(ef_newvalue>envelope_follower)
-                        ef_rate = zone->ef_attack;
-                else
-                        ef_rate = zone->ef_release;
+            if(ef_newvalue>envelope_follower)
+                    ef_rate = zone->ef_attack;
+            else
+                    ef_rate = zone->ef_release;
 
-                //float x = pi * powf(2,-ef_rate) * integratorconst;
-                float x = pi * note_to_pitch(-12*ef_rate) * integratorconst;
-                float tm = 2-cos(x);
-                float p = tm - sqrt(tm*tm - 1);
+            //float x = pi * powf(2,-ef_rate) * integratorconst;
+            float x = pi * note_to_pitch(-12*ef_rate) * integratorconst;
+            float tm = 2-cos(x);
+            float p = tm - sqrt(tm*tm - 1);
 
-                envelope_follower = envelope_follower*p + (1-p)*ef_newvalue;
-        }*/
+            envelope_follower = envelope_follower*p + (1-p)*ef_newvalue;
+    }*/
 
-        _MM_ALIGN16 float tempbuf[2][block_size],
-        postfader_buf[2][block_size];
+    _MM_ALIGN16 float tempbuf[2][block_size], postfader_buf[2][block_size];
 
     if (use_stereo)
     {
@@ -649,9 +653,9 @@ bool sampler_voice::process_block(float *p_L, float *p_R, float *p_aux1L, float 
         copy_block(output[0], output[1], block_size_quad);
     }
 
-    perfslot(9)
+    perfslot(9);
 
-        const unsigned int bufof = block_size;
+    const unsigned int bufof = block_size;
     vca.multiply_2_blocks(output[0], output[1], block_size_quad);
     faderL.multiply_block_to(output[0], postfader_buf[0], block_size_quad);
     faderR.multiply_block_to(output[1], postfader_buf[1], block_size_quad);

--- a/src/sampler_voice.h
+++ b/src/sampler_voice.h
@@ -91,7 +91,7 @@ class sampler_voice
     float envelope_follower, fpitch;
     float lag[2];
 
-    inline int get_filter_type(int id);
+    inline int get_filter_type(int id) const;
     // void filter_ctrldata(int);
     bool sample_started, first_run;
     bool upsampling;

--- a/src/synthesis/envelope.cpp
+++ b/src/synthesis/envelope.cpp
@@ -62,13 +62,13 @@ void Envelope::Assign(float *envA, float *envH, float *envD, float *envS, float 
     R = envR;
     shape = envshape;
 
-    this->edata = edata;
+    // this->edata = edata;
 
     state = sIdle;
     level = 0.;
     output = 0.;
     block = 1; // counter
-    this->is_AEG = is_AEG;
+    // this->is_AEG = is_AEG;
 }
 
 //-------------------------------------------------------------------------------------------------

--- a/src/windows_compat.h
+++ b/src/windows_compat.h
@@ -182,8 +182,6 @@ inline bool SHGetPathFromIDList(LPITEMIDLIST, const char *) { return false; }
 inline HANDLE FindFirstFile(const char *, WIN32_FIND_DATA *) { return 0; }
 inline HANDLE FindNextFile(HANDLE, WIN32_FIND_DATA *) { return 0; }
 
-inline void OutputDebugString(const char *c) { std::cout << c << std::endl; }
-
 #define MAX_PATH 1024
 
 #ifndef _MM_ALIGN16

--- a/unused/thread/threadsafety.cpp
+++ b/unused/thread/threadsafety.cpp
@@ -4,6 +4,8 @@
 #include <CoreServices/CoreServices.h>
 #endif
 
+#error THIS IS NO LONGER NEEDED
+
 c_sec::c_sec()
 {
 	refcount = 0;

--- a/unused/thread/threadsafety.h
+++ b/unused/thread/threadsafety.h
@@ -1,5 +1,7 @@
 #pragma once
 
+#error THIS IS NO LONGER NEEDED
+
 #if MAC
 #include <CoreServices/CoreServices.h>
 #elif LINUX

